### PR TITLE
feat(server): NoProtocolInHostRule operates on Context

### PR DESCRIPTION
**Pull request by roxspring**
- Rule updated to operate on `Context` and produce violations with a pointer
- Continues to block `Swagger` hosts with protocols since ["This MUST be the host only and does not include the scheme nor sub-paths"](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#swagger-object)
- Allows equivalent  `OpenAPI` server urls with protocols since the spec [clearly expects a full URL](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#server-object)

Relates to #834

### CODE REVIEW

File: server/src/test/java/de/zalando/zally/rule/zally/NoProtocolInHostRuleTest.kt [hunk 1]
```diff
@@ -1,54 +1,99 @@
...
+            .descriptionsEqualTo("'https://google.com' contains protocol information which should be listed separately as schemes")
+            .pointersEqualTo("/host")
     }
 
     @Test
-    fun negativeCaseHttps() {
-        val swagger = Swagger().apply { host = "https://google.com" }
-        val res = rule.validate(swagger)
-        assertThat(res?.copy(description = "")).isEqualTo(expectedViolation)
+    fun `validate swagger with SPP json returns no violations`() {
```
Comment (maxim-tschumak) [inline server/src/test/java/de/zalando/zally/rule/zally/NoProtocolInHostRuleTest.kt]: Let's remove `validate swagger with SPP json returns no violations` and `validate swagger with SPA yaml returns no violations` tests. I don't see much benefit in them and think we should avoid having full-size API specifications as test fixtures.

File: server/src/main/java/de/zalando/zally/rule/zally/NoProtocolInHostRule.kt [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
-    private val desc = "Information about protocol should be placed in schema. Current host value '%s' violates this rule"
 
     @Check(severity = Severity.MUST)
-    fun validate(swagger: Swagger): Violation? {
-        val host = swagger.host.orEmpty()
-        return if ("://" in host)
-            Violation(desc.format(host), emptyList())
-        else null
+    fun validate(context: Context): List<Violation> {
+        val host = context.swagger?.host.orEmpty()
```
Comment (maxim-tschumak) [inline server/src/main/java/de/zalando/zally/rule/zally/NoProtocolInHostRule.kt]: The fact that only `Swagger` specifications a checked is a bit hidden. Should we use `Context#isOpenAPI3()` to make it more explicit?

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by ddufourboivin-zalando** [comment]
👍

**Comment by ddufourboivin-zalando** [comment]
Thanks for that, @roxspring !

**Pull request merged**